Test failure on Arm64

Unfortunately I can't debug since I don't have such a machine set up; the tests were run by Hydra, the CI server for Nix/Nixpkgs for [this PR](https://github.com/NixOS/nixpkgs/pull/77472).

Release 3.0.0.

```
FAIL: nibabel.tests.test_volumeutils.test_a2f_nan2zero_range
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/nix/store/iqysj9ccwz5xww339z07jyl6k5zc5074-python3.7-nose-1.3.7/lib/python3.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/build/nibabel-3.0.0/nibabel/tests/test_volumeutils.py", line 675, in test_a2f_nan2zero_range
    assert_array_equal([-128, -128, -128, nan_cast], back_arr)
  File "/nix/store/7k6dbm0p5imsnivp1xshbzkj30i28cjq-python3.7-numpy-1.18.0/lib/python3.7/site-packages/numpy/testing/_private/utils.py", line 936, in assert_array_equal
    verbose=verbose, header='Arrays are not equal')
  File "/nix/store/7k6dbm0p5imsnivp1xshbzkj30i28cjq-python3.7-numpy-1.18.0/lib/python3.7/site-packages/numpy/testing/_private/utils.py", line 846, in assert_array_compare
    raise AssertionError(msg)
AssertionError:
Arrays are not equal
Mismatched elements: 1 / 4 (25%)
Max absolute difference: 1
Max relative difference: 1.0
 x: array([-128, -128, -128,    0])
 y: array([-128, -128, -128,   -1], dtype=int8)
----------------------------------------------------------------------
Ran 7671 tests in 227.364s
```

I ran nibabel's tests on arm64, and only one failed.

```
FAILED tests/test_volumeutils.py::test_a2f_nan2zero_range - AssertionError: 
1 failed, 4784 passed, 111 skipped, 6 xfailed, 5662 warnings in 2153.20s (0:35:53)
```

To reproduce, run nibabel in an arm64 docker image:

```
# Register qemu interpreter things to be able to run arm64 etc on amd64
docker run --rm --privileged docker/binfmt:820fdd95a9972a5308930a2bdfb8573dd4447ad3
docker run --rm -it -w /tmp/for-testing --platform linux/aarch64 kaczmarj/nibabel:latest
pytest --doctest-modules -v --pyargs nibabel
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -29,6 +29,8 @@ jobs:
   include:
     - arch: arm64
       python: 3.6
+      env:
+        - DEPENDS="numpy"
     # Basic dependencies only
     - python: 3.6
       env:

--- a/.travis.yml
+++ b/.travis.yml
@@ -5,6 +5,7 @@
 # can't be any leading "-"s - All newlines will be removed, so use ";"s
 
 os: linux
+arch: amd64
 dist: xenial
 language: python
 cache: pip
@@ -26,6 +27,8 @@ python:
 
 jobs:
   include:
+    - arch: arm64
+      python: 3.6
     # Basic dependencies only
     - python: 3.6
       env:

--- a/.travis.yml
+++ b/.travis.yml
@@ -17,9 +17,9 @@ env:
         - INSTALL_TYPE="setup"
         - CHECK_TYPE="test"
         - EXTRA_WHEELS="https://3f23b170c54c2533c070-1c8a9b3114517dc5fe17b7c3f8c63a43.ssl.cf2.rackcdn.com"
-        - PRE_WHEELS="https://7933911d6844c6c53a7d-47bd50c35cd79bd838daf386af554a83.ssl.cf2.rackcdn.com"
+        - NIGHTLY_WHEELS="https://pypi.anaconda.org/scipy-wheels-nightly/simple"
         - EXTRA_PIP_FLAGS="--find-links=$EXTRA_WHEELS"
-        - PRE_PIP_FLAGS="--pre $EXTRA_PIP_FLAGS --find-links $PRE_WHEELS"
+        - PRE_PIP_FLAGS="--pre $EXTRA_PIP_FLAGS --extra-index-url $NIGHTLY_WHEELS"
 
 python:
     - 3.7
@@ -31,6 +31,7 @@ jobs:
       python: 3.6
       env:
         - DEPENDS="numpy"
+        - EXTRA_PIP_FLAGS="$PRE_PIP_FLAGS"
     # Basic dependencies only
     - python: 3.6
       env:

--- a/nibabel/tests/test_volumeutils.py
+++ b/nibabel/tests/test_volumeutils.py
@@ -678,7 +678,7 @@ def test_a2f_nan2zero_range():
         with pytest.raises(ValueError):
             write_return(arr_no_nan, fobj, np.int8, intercept=129)
         # OK with nan2zero false, but we get whatever nan casts to
-        nan_cast = np.array(np.nan).astype(np.int8)
+        nan_cast = np.array(np.nan, dtype=dt).astype(np.int8)
         back_arr = write_return(arr, fobj, np.int8, intercept=129, nan2zero=False)
         assert_array_equal([-128, -128, -128, nan_cast], back_arr)
         # divslope

